src/ResizeObserverController.js

```diff
@@ -37,25 +55,30 @@ export default class ResizeObserverController {
     mutationsObserver_ = null;
 
     /**
-     * A list of connected observers.
+     * Monitors the shadow root host for size changes.
      *
-     * @private {Array<ResizeObserverSPI>}
+     * @private {ResizeObserverSPI}
      */
-    observers_ = [];
+    hostObserver_ = null;
 
     /**
-     * Holds reference to the controller's instance.
+     * A list of connected observers.
      *
-     * @private {ResizeObserverController}
+     * @private {Array<ResizeObserverSPI>}
      */
-    static instance_ = null;
+    observers_ = [];
 
     /**
      * Creates a new instance of ResizeObserverController.
      *
      * @private
+     * @param {Node} rootNode - The root node that this controller monitors.
+     * @param {GlobalResizeObserverController} globalController - The global
+     * controller for all roots.
      */
-    constructor() {
+    constructor(rootNode, globalController) {
```

Comment:
whats the reason for `globalController` as an arg vs. always using the singleton getter? Is it for DI/testability?

Reply:
No. It's just for import-dependency-chain. Otherwise the deps will cause a dep loop.